### PR TITLE
[Docs] Add an ionogram example in MMYOLO application

### PR DESCRIPTION
**Pull request by VoyagerXvoyagerx**
## Motivation

Add MMYOLO application examples.

## Modification

- Add config files for MMYOLO application examples to  `/projects/misc/ionogram_detection/`
- Add doc MMYOLO 应用范例介绍 in Chinese
- Update the link for industry examples in `README.md`
- Update the doc link in [/projects/misc/custom_dataset/README.md](https://github.com/open-mmlab/mmyolo/tree/dev/projects/misc/custom_dataset) 

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
+# 基于 MMYOLO 的频高图实时目标检测 benchmark
+
+## 数据集构建
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:3]: 这个地方要写清楚，用户下载数据集后，应该要解压，并且放置到某个路径下？ 如果数据格式需要调整，需要说明调整前面文件夹结构

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+├── dataset_analysis.ipynb
+├── dataset.ipynb
+├── images
+├── labels
+├── test_images
+├── train_images
+└── val_images
+```
+
+使用以下代码可以统计数据集中各类别的实例数量：
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:67]: 直接用这个 tools/analysis_tools/dataset_analysis.py 可以吗？

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+0  311  101  137  303  626       20
+
+test 646 images
+     E  Esl  Esc   F1   F2  Fspread
+0  293  123  127  297  623       22
+```
+
+## 配置文件
+
+配置文件在目录 `/configs/custom_dataset` 下。
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:115]:
```suggestion
配置文件在目录 `/configs/custom_dataset/ionogram_detection` 下。
```

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+根据配置文件中的 `train_pipeline`，训练时采用的数据增强策略包括：
+
+- 马赛克增强
+- 仿射变换
+- Albumentations 数据增强工具包（包括多种数字图像处理方法）
+- HSV随机增强图像
+- 随机水平翻转
+
+修改 `train_pipeline` 并可视化不同的数据增强方法：
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:145]: 这一步是干啥？如果想可视化每个pipeline输出效果，可以直接用 tools/analysis_tools/browse_dataset.py 脚本

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+Model Parameters: 7.036M
+==============================
+```
+
+## 训练和测试
+
+1. 训练
+
+```shell
+python tools/train.py configs/custom_dataset/yolov5/yolov5_s-v61_syncbn_fast_1xb96-100e_ionogram.py
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:191]: 配置路径错误

File: projects/misc/ionogram_detection/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Tips: 这是 MMYOLO 应用范例的配置文件，请结合 [基于 MMYOLO 的频高图实时目标检测 benchmark](/docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md) 来使用。
```
[the remark this answers is not in the document]
Comment (VoyagerXvoyagerx) [reply projects/misc/ionogram_detection/README.md:1]: 我在 fork 的仓库里点击，是可以跳转的。[https://github.com/VoyagerXvoyagerx/mmyolo/tree/dev/projects/misc/ionogram_detection](https://github.com/VoyagerXvoyagerx/mmyolo/tree/dev/projects/misc/ionogram_detection)

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md (hunk 2 above)
Comment (VoyagerXvoyagerx) [reply docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:67]: 可以，统计 bbox 数量的内容放在配置文件部分了，因为使用数据集分析脚本需要先修改好配置文件

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+使用下面的命令可以将 COCO 的 label 在图片上进行显示，这一步可以验证刚刚转换是否有问题。
+
+```shell
+python tools/analysis_tools/browse_coco_json.py --img-dir ./Iono4311/images \
+                                                --ann-file ./Iono4311/annotations/annotations_all.json
+```
+
+4. 划分训练集、验证集、测试集
+
+设置70%的图片为训练集，15%作为验证集，15%为测试集。
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:49]:
```suggestion
设置 70% 的图片为训练集，15% 作为验证集，15% 为测试集。
```

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+Model Flops: 7.947G
+Model Parameters: 7.036M
+==============================
+```
+
+## 训练和测试
+
+1. 训练
+
+训练可视化：本范例按照[标注+训练+测试+部署全流程](/pr/mmyolo/docs/zh_cn/recommended_topics/labeling_to_deployment_tutorials.md#91-训练可视化)中的步骤安装和配置 [wandb](https://wandb.ai/site)。
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:178]:
```suggestion
训练可视化：本范例按照[标注+训练+测试+部署全流程](https://mmyolo.readthedocs.io/zh_CN/dev/recommended_topics/labeling_to_deployment_tutorials.html#id11)中的步骤安装和配置 [wandb](https://wandb.ai/site)。
```

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+python tools/train.py projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py
+```
+
+2. 测试
+
+指定配置文件和模型的路径以启动测试：
+
+```shell
+python tools/test.py projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py \
+                     work_dirs/yolov5_s_100e/best_coco-test-0.584.pth
```
Comment (hhaAndroid) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:213]:
```suggestion
                     work_dirs/yolov5_s-v61_fast_1xb96-100e_ionogram/xxx
```
best_coco-test-0.584.pth 这个权重名不对吧

File: projects/misc/ionogram_detection/yolov7/yolov7_l_fast_1xb16-100e_ionogram.py [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+train_num_workers = 8
+
+# -----model related-----
+anchors = [[[14, 14], [35, 6], [32, 18]], [[32, 45], [28, 97], [52, 80]],
+           [[71, 122], [185, 94], [164, 134]]]
+
+# -----train val related-----
+#  base_lr_default * (your_bs 32 / default_bs (8 x 16))
+base_lr = _base_.base_lr * train_batch_size_per_gpu / (8 * 16)
+load_from = './work_dirs/yolov7_l_syncbn_fast_8x16b-300e_coco_20221123_023601-8113c0eb.pth'  # noqa
```
Comment (hhaAndroid) [inline projects/misc/ionogram_detection/yolov7/yolov7_l_fast_1xb16-100e_ionogram.py:30]: 写 url 就行

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+</div>
+
+1. 数据集准备
+
+下载数据后，放置在 MMYOLO 仓库的根目录下，使用 `unzip test.zip` 命令（linux）解压至当前文件夹。解压后的文件夹结构为：
+
+```shell
+Iono4311/
+├── images
+└── labels
```
Comment (RangeKing) [inline docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:23]:
```
Iono4311/
├── images
|      ├── example.jpg
|      └── ...
└── labels
       └── annotations_all.json
```
modify the files tree like this.

File: docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md [hunk 12]
```diff
@@ -0,0 +1,301 @@
+# 基于 MMYOLO 的频高图实时目标检测 benchmark
+
+## 数据集构建
+
+数字频高图是获取电离层实时信息最重要的途径。电离层结构检测对精准提取电离层关键参数，具有非常重要的研究意义。
+
+利用中国科学院在海南、武汉、怀来获取的不同季节的4311张频高图建立数据集，人工标注出 E 层、Es-c 层、Es-l 层、F1 层、F2 层、Spread F 层共 6 种结构。[数据集下载](https://github.com/VoyagerXvoyagerx/Ionogram_detection/releases/download/Dataset/Iono4311.zip)
```
[the remark this answers is not in the document]
Comment (RangeKing) [reply docs/zh_cn/recommended_topics/application_examples/ionogram_detection.md:7]: Check out for spaces between [English, numbers] and Chinese.

File: projects/misc/ionogram_detection/rtmdet/rtmdet_l_fast_1xb32-100e_ionogram.py [hunk 13]
```diff
@@ -0,0 +1,110 @@
+_base_ = '../../../../configs/rtmdet/rtmdet_l_syncbn_fast_8xb32-300e_coco.py'
```
Comment (RangeKing) [inline projects/misc/ionogram_detection/rtmdet/rtmdet_l_fast_1xb32-100e_ionogram.py:1]:
```suggestion
_base_ = 'mmyolo::rtmdet/rtmdet_l_syncbn_fast_8xb32-300e_coco.py'
```

File: projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
+_base_ = '../../../../configs/yolov5/yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py'  # noqa
```
Comment (RangeKing) [inline projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py:1]:
```suggestion
_base_ = 'mmyolo::yolov5/yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py'  # noqa
```

File: projects/misc/ionogram_detection/yolov6/yolov6_m_fast_1xb32-100e_ionogram.py [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+# The scaling factor that controls the depth of the network structure
+deepen_factor = 0.6
+# The scaling factor that controls the width of the network structure
+widen_factor = 0.75
+
+# -----train val related-----
+affine_scale = 0.9  # YOLOv5RandomAffine scaling ratio
+load_from = 'https://download.openmmlab.com/mmyolo/v0/yolov6/yolov6_m_syncbn_fast_8xb32-300e_coco/yolov6_m_syncbn_fast_8xb32-300e_coco_20221109_182658-85bda3f4.pth'  # noqa
+
+# ============================== Unmodified in most cases ===================
```
Comment (RangeKing) [inline projects/misc/ionogram_detection/yolov6/yolov6_m_fast_1xb32-100e_ionogram.py:14]:
```suggestion
# ====================== Unmodified in most cases ===================
```

File: projects/misc/ionogram_detection/yolov6/yolov6_l_fast_1xb32-100e_ionogram.py [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+
+# ======================= Modified parameters =======================
+# -----model related-----
+deepen_factor = 1
+widen_factor = 1
+
+# -----train val related-----
+load_from = 'https://download.openmmlab.com/mmyolo/v0/yolov6/yolov6_l_syncbn_fast_8xb32-300e_coco/yolov6_l_syncbn_fast_8xb32-300e_coco_20221109_183156-91e3c447.pth'  # noqa
+
+# ============================== Unmodified in most cases ===================
```
Comment (RangeKing) [inline projects/misc/ionogram_detection/yolov6/yolov6_l_fast_1xb32-100e_ionogram.py:11]:
```suggestion
# ====================== Unmodified in most cases ===================
```

File: projects/misc/ionogram_detection/yolov7/yolov7_l_fast_1xb16-100e_ionogram.py [hunk 17]
```diff
@@ -0,0 +1,98 @@
+_base_ = '../../../../configs/yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py'
```
Comment (RangeKing) [inline projects/misc/ionogram_detection/yolov7/yolov7_l_fast_1xb16-100e_ionogram.py:1]:
```suggestion
_base_ = 'mmyolo::yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py'
```

**Review by hhaAndroid — APPROVED** [review]

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 18]
```diff
@@ -0,0 +1,307 @@
...
+
+Use the script `tools/dataset_converters/labelme2coco.py` to convert labelme labels to COCO labels.
+
+```shell
+python tools/dataset_converters/labelme2coco.py --img-dir ./Iono4311/images \
+                                                --labels-dir ./Iono4311/labels \
+                                                --out ./Iono4311/annotations/annotations_all.json
+```
+
+3. Check the converted COCO label
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:43]:
```suggestion
3. Check the converted COCO labels
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 19]
```diff
@@ -0,0 +1,307 @@
...
+
+Use the following command to display the COCO labels on the image, which verifies that the conversion has been problem-free.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:45]:
```suggestion
To confirm that the conversion process went successfully, use the following command to display the COCO labels on the images.
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 20]
```diff
@@ -0,0 +1,307 @@
...
+| E             | 98           |
+| Es-l          | 27           |
+| Es-c          | 46           |
+| F1            | 100          |
+| F2            | 194          |
+| Spread-F      | 6            |
++---------------+--------------+
+```
+
+This indicate that the distribution of catogories in the dataset is unbalanced.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:113]:
```suggestion
This indicates that the distribution of categories in the dataset is unbalanced.
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 21]
```diff
@@ -0,0 +1,307 @@
...
+python tools/analysis_tools/optimize_anchors.py projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py \
+                                                --algorithm v5-k-means \
+                                                --input-shape 640 640 \
+                                                --prior-match-thr 4.0 \
+                                                --out-dir work_dirs/dataset_analysis_5_s
+```
+
+4. Model complexity analysis
+
+With the config file, the params and flops can be calculated by the script `tools/analysis_tools/get_flops.py`. Take yolov5-s as an example:
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:163]:
```suggestion
With the config file, the parameters and FLOPs can be calculated by the script `tools/analysis_tools/get_flops.py`. Take yolov5-s as an example:
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 22]
```diff
@@ -0,0 +1,307 @@
...
+
+```shell
+python tools/analysis_tools/get_flops.py projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py
+```
+
+The following output indicates that the model has 7.947G flops with the input shape (640, 640), and a total of 7.036M learnable parameters.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:169]:
```suggestion
The following output indicates that the model has 7.947G FLOPs with the input shape (640, 640), and a total of 7.036M learnable parameters.
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 23]
```diff
@@ -0,0 +1,307 @@
...
+python tools/test.py projects/misc/ionogram_detection/yolov5/yolov5_s-v61_fast_1xb96-100e_ionogram.py \
+                     work_dirs/yolov5_s-v61_fast_1xb96-100e_ionogram/xxx
+```
+
+## Experiments and results
+
+### Choose a suitable batch size
+
+- Often, the batch size governs the training speed, and the ideal batch size will be the largest batch size supported by the available hardware.
+- When the accelerators aren't yet saturated, if the batch size doubles, the training throughput should also double (or at least nearly double). Equivalently, the time per step should be constant (or at least nearly constant) as the batch size increases.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:226]:
```suggestion
- If the video memory is not yet fully utilized, doubling the batch size should result in a corresponding doubling (or close to doubling) of the training throughput. This is equivalent to maintaining a constant (or nearly constant) time per step as the batch size increases.
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 24]
```diff
@@ -0,0 +1,307 @@
...
+| YOLOv5-s | 200(188)    | True  | 96        | **8**       | 54.66%           | 67 min        | 0.576   |
+
+<div align=center>
+<img width="60%" src="https://user-images.githubusercontent.com/67947949/223635966-948c8424-1ba8-4df0-92d7-079029dc1231.png">
+
+The proportion of data loading time to the total time of each step.
+
+</div>
+
+Based on the results above, we can condlude that
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:253]:
```suggestion
Based on the results above, we can conclude that
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 25]
```diff
@@ -0,0 +1,307 @@
...
+
+- AMP has little impact on the accuracy of the model, but can significantly reduce the memory usage while training.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:255]:
```suggestion
- AMP has little impact on the accuracy of the model, but can significantly reduce memory usage while training.
```

File: docs/en/recommended_topics/application_examples/ionogram_detection.md [hunk 26]
```diff
@@ -0,0 +1,307 @@
...
+| Val mAP    | 0.507                                                                                            | 0.550                                                                                              | 0.572                                                                                                     | 0.567                                                                                                              | 0.575                                                                                       |
+
+The results indicate that mosaic augmentation and random affine transformation can significantly improve the performance on the validation set.
+
+#### Using pre-trained models
+
+To not use pre-trained weights, simply set `load_from = None` in the config file. For experiments that do not use pre-trained weights, increase the base learning rate by four times and increase the number of training epochs to 200 to ensure the model is sufficiently trained.
```
Comment (RangeKing) [inline docs/en/recommended_topics/application_examples/ionogram_detection.md:277]:
```suggestion
If you prefer not to use pre-trained weights, you can simply set `load_from = None` in the config file. For experiments that do not use pre-trained weights, it is recommended to increase the base learning rate by a factor of four and extend the number of training epochs to 200 to ensure adequate model training.
```

**Pull request merged**